lib/modules/manager/kotlin-script/extract.ts

```diff
@@ -0,0 +1,35 @@
+import { regEx } from '../../../util/regex';
+import { MavenDatasource } from '../../datasource/maven';
+import type { PackageDependency, PackageFile } from '../types';
+
+const dependsOnRegex = regEx(
+  /@file\s*:\s*DependsOn\s*\(\s*(?<replaceString>"(?<groupId>.*):(?<artifactId>.*):(?<version>.*)")\s*\)/g
+);
+const repositoryRegex = regEx(
+  /@file\s*:\s*Repository\s*\(\s*"(?<repositoryName>.*)"\s*\)/g
+);
+
+export function extractPackageFile(fileContent: string): PackageFile | null {
+  const registryUrls: string[] = [...fileContent.matchAll(repositoryRegex)]
+    .map((match) => {
+      return match.groups?.repositoryName;
+    })
+    .filter((x): x is string => x !== null);
+
+  const deps: PackageDependency[] = [
+    ...fileContent.matchAll(dependsOnRegex),
+  ].map((match) => {
```

Comment:
use `for...of` loop instead

Reply:
I'm not a fan of imperative style, but I'll respect your style :)

Reply:
The imperative style is much more readable, especilly when the callback functions are huge. They are ok for small / simple callbacks. 🤗 

https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md